helix-view/src/editor.rs

```diff
@@ -67,20 +72,38 @@ impl Editor {
             theme,
             language_servers,
             registers: Registers::default(),
-            status_msg: None,
+            status: String::new(),
+            severity: None,
         }
     }
 
+    /// Get immutable status. Should check severity to see if status should be empty.
+    pub fn status(&self) -> &str {
+        &self.status
+    }
```

Comment:
What about this?
```rust
pub fn status(&self) -> Option<(&str, Severity)> {
    self.severity.map(|level| (&self.status, level))
}
```
If `status` should only be used if `severity` is `Some`, we might as well enforce it in the API.

Reply:
IIRC I tried that and get reborrow error or something.